File: Java/Bit_Manipulation/_09_Subsets.java
// 09. Write a program to generate all possible subsets of a set {7,8,9,6}.

import java.util.Arrays;

public class _09_Subsets {
    private static void printSubsets(int arr[]) {
        for (int i = 0; i < (1 << arr.length); i++) {
            System.out.print("{ ");
            for (int j = 0; j < arr.length; j++)
                if (0 != (i & (1 << j)))
                    System.out.print(arr[j] + " ");
            System.out.println("}");
        }
    }

    public static void main(String[] args) {
        int[] arr = { 7, 8, 9, 6 };
        System.out.println("All possible subsets of " + Arrays.toString(arr) + " 👇");
        printSubsets(arr);
    }
}

File: C++/Bit_Manipulation/_09_Subsets..cpp
// 09. Write a program to generate all possible subsets of a set {7,8,9,6}.
#include <iostream>
using namespace std; // for compatibility  -  with earlier Versions
void printSubsets(int arr[], int size)
{
    for (int i = 0; i < (1 << size); i++)
    {
        printf("{ ");
        for (int j = 0; j < size; j++)
            if (0 != (i & (1 << j)))
                printf("%d ", arr[j]);
        printf("}\n");
    }
}
int main()
{
    int arr[] = {7, 8, 9, 6};
    int size = sizeof(arr) / sizeof(arr[0]);
    printSubsets(arr, size);
}